miner/worker.go

```diff
@@ -456,7 +464,11 @@ func (w *worker) mainLoop() {
 					continue
 				}
 				w.mu.RLock()
-				coinbase := w.coinbase
+				txFeeRecipient := w.txFeeRecipient
+				if !w.chainConfig.IsDonut(w.current.header.Number) && w.txFeeRecipient != w.validator {
```

Comment:
do we know if current.header != nil?

Reply:
The only function that sets the `current` is `makeCurrent`, which is called only by the `Prepare` function after initialising the header.
I can't use the `header.Coinbase` in that line, because it's the case where the consensus is not running, and leaves the `header.Coinbase` with `nil`